### PR TITLE
Bug 1834723: Add e2e validation for cluster machine approver cluster operator status

### PR DESCRIPTION
**Pull request by enxebre**
Since we added support for the machine approver cluster operator, we want to validate it works as expected.
https://github.com/openshift/cluster-machine-approver/pull/71

### CODE REVIEW

File: pkg/operators/cluster-machine-approver.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+const (
+	cmaDeployment      = "machine-approver"
+	cmaClusterOperator = "machine-approver"
+)
+
+var _ = Describe("[Feature:Operators] Cluster Machine Approver deployment", func() {
+	var client runtimeclient.Client
+	var err error
+	client, err = framework.LoadClient()
```
Comment (JoelSpeed) [inline pkg/operators/cluster-machine-approver.go:18]: Any reason not to inline the declaration? Or should this be in a before each or something?
```suggestion
	client, err := framework.LoadClient()
```

File: pkg/operators/cluster-machine-approver.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	It("should be available", func() {
+		Expect(framework.IsDeploymentAvailable(client, cmaDeployment)).To(BeTrue())
+	})
+})
+
+var _ = Describe("[Feature:Operators] Cluster Machine Approver Cluster Operator Status", func() {
+	var client runtimeclient.Client
+	var err error
+	client, err = framework.LoadClient()
```
Comment (JoelSpeed) [inline pkg/operators/cluster-machine-approver.go:29]: Ditto
```suggestion
	client, err := framework.LoadClient()
```

File: pkg/operators/cluster-machine-approver.go (hunk 1 above)
Comment (enxebre) [reply pkg/operators/cluster-machine-approver.go]: no reason indeed. Thanks!

File: pkg/operators/cluster-machine-approver.go (hunk 2 above)
Comment (enxebre) [reply pkg/operators/cluster-machine-approver.go]: no reason indeed. Thanks!

File: pkg/operators/cluster-machine-approver.go (hunk 1 above)
Comment (enxebre) [reply pkg/operators/cluster-machine-approver.go]: inlined it, let me know if it looks good now.

**Comment by JoelSpeed** [comment]
/lgtm cancel

Just had a look at the tests to check, I think it's broken because of this

File: pkg/operators/cluster-machine-approver.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+const (
+	cmaDeployment      = "machine-approver"
+	cmaClusterOperator = "machine-approver"
+)
+
+var _ = Describe("[Feature:Operators] Cluster Machine Approver deployment", func() {
+	client, err := framework.LoadClient()
+	Expect(err).NotTo(HaveOccurred())
```
Comment (JoelSpeed) [inline pkg/operators/cluster-machine-approver.go:16]: I think this needs to be inside a BeforeEach

Comment (enxebre) [reply pkg/operators/cluster-machine-approver.go]: Moved it within the it()

**Comment by JoelSpeed** [comment]
/lgtm

Assuming tests pass

**Comment by enxebre** [comment]
test were failing to look up the mca deployment. I added https://github.com/openshift/cluster-api-actuator-pkg/pull/152/commits/eddcc917b6177aa1aa7654b5b9f4c0211e54cdd2

**Comment by enxebre** [comment]
@JoelSpeed PTAL

**Comment by enxebre** [comment]
>Passing an empty string for the namespace seems a bit weird to me, any reason not to make it mandatory to pass a namespace, we have a constant for it already?

sg. Done.

File: pkg/operators/machine-api-operator.go [hunk 4]
```diff
@@ -28,22 +28,22 @@ var _ = Describe("[Feature:Operators] Machine API operator deployment should", f
 		Expect(err).NotTo(HaveOccurred())
 
 		deploymentName := "machine-api-controllers"
-		initialDeployment, err := framework.GetDeployment(client, deploymentName)
+		initialDeployment, err := framework.GetDeployment(client, deploymentName, framework.MachineAPINamespace)
 		if err != nil {
-			initialDeployment, err = framework.GetDeployment(client, deploymentDeprecatedName)
+			initialDeployment, err = framework.GetDeployment(client, deploymentDeprecatedName, "")
```
Comment (JoelSpeed) [inline pkg/operators/machine-api-operator.go:33]:
```suggestion
			initialDeployment, err = framework.GetDeployment(client, deploymentDeprecatedName, framework.MachineAPINamespace)
```

Comment (enxebre) [reply pkg/operators/machine-api-operator.go]: thanks! I actually dropped the deprecated deployment check now https://github.com/openshift/cluster-api-actuator-pkg/pull/152/commits/f19ea2894ea68272a794a6b9aa3ca817acb6e751

**Comment by enxebre** [comment]
https://github.com/openshift/cluster-api-actuator-pkg/pull/152#discussion_r424971124
@elmiko @JoelSpeed PTAL

**Comment by elmiko** [comment]
thanks Alberto!
/approved

**Pull request merged**